### ISSUE TITLE
3.3.9 Accessible Authentication (Enhanced) (Level AAA)

### ISSUE THREAD

**Issue by maryjom** [w3c/wcag2ict#565]
#### [3.3.9 Accessible Authentication (Enhanced)](https://www.w3.org/TR/WCAG22/#accessible-authentication-enhanced)

(Level AAA)

>A [cognitive function test](https://www.w3.org/TR/WCAG22/#dfn-cognitive-function-test) (such as remembering a password or solving a puzzle) is not required for any step in an authentication [process](https://www.w3.org/TR/WCAG22/#dfn-processes) unless that step provides at least one of the following:
><dl>
><dt>Alternative</dt>
><dd>Another authentication method that does not rely on a cognitive function test.</dd>
><dt>Mechanism</dt>
><dd>A [mechanism](https://www.w3.org/TR/WCAG22/#dfn-mechanism) is available to assist the user in completing the cognitive function test.</dd>
></dl>

##### Applying SC 3.3.9 Accessible Authentication (Enhanced) to Non-Web Documents and Software
This applies directly as written, and as described in [Intent from Understanding Success Criterion 3.3.9](https://www.w3.org/WAI/WCAG22/Understanding/accessible-authentication-enhanced.html#intent).

**NOTE (Added)** @@ May need notes. @@

***
**For WCAG2ICT consideration:** 
- Are notes needed for non-web software and documents?
- Are there considerations for closed functionality?
***

**Comment by Lboniello** [comment]
No changes needed to the overall content, but would suggest editorial change:

"Applying SC 3.3.9 Accessible Authentication (Enhanced) to Non-Web Documents and Software"
 
becomes:

"Applying SC 3.3.9 Accessible Authentication (Enhanced) to Non-Web Documents and Non-Web Software"

Some considerations for discussion. Examples in this SC are problematic for closed systems.  A third party password manager is not used in a closed system. Web auth alternatives that rely on biometrics will require a second biometric option, using different biometric features.  Two factor authentication can not be made available via a USB on a closed system. Two factor authentication via QR codes can be problematic for users who are blind/have low vision.  Because these are examples, there is no change to the SC itself but do we need to note anything to show more accessible alternatives and/or draw attention to limitations of closed systems?

**Comment by pday1** [comment]
Hi @Lboniello,
I've modified it slightly as I think you were proposing content for SC problematic for closed. Feel free to edit the SC problematic section - I just took the bits that were not questions. 

Proposed text is therefore as follows. 

> Applying SC 3.3.9 Accessible Authentication (Enhanced) to non-web documents and non-web software
> 
> This applies directly as written, and as described in [Intent from Understanding Success Criterion 3.3.9](https://www.w3.org/WAI/WCAG22/Understanding/accessible-authentication-enhanced.html#intent).
> 
> NOTE (FOR NON-WEB SOFTWARE)
> See also the [Comments on Closed Functionality](https://w3c.github.io/wcag2ict/#comments-on-closed-functionality).

Suggested content for SC in Success Criteria Problematic for Closed Functionality

> Examples in this SC are problematic for closed systems. A third party password manager is not used in a closed system. Web auth alternatives that rely on biometrics will require a second biometric option, using different biometric features. Two factor authentication can not be made available via a USB on a closed system. Two factor authentication via QR codes can be problematic for users who are blind/have low vision.

**Comment by pday1** [comment]
From discussion in TF meeting 23 April 

Make consistent with 3.3.8 - re-use that rather than adding new notes.

**Comment by Lboniello** [comment]
> Proposed text is as follows.
> 
> > Applying SC 3.3.9 Accessible Authentication (Enhanced) to non-web documents and non-web software
> > This applies directly as written, and as described in [Intent from Understanding Success Criterion 3.3.9](https://www.w3.org/WAI/WCAG22/Understanding/accessible-authentication-enhanced.html#intent).
> > NOTE (FOR NON-WEB SOFTWARE)
> > See also the [Comments on Closed Functionality](https://w3c.github.io/wcag2ict/#comments-on-closed-functionality).
> 
> Suggested content for SC in Success Criteria Problematic for Closed Functionality
> 
Note 1
"Object recognition" and "Personal content" may be represented by images, video, or audio.

Note 2
Examples of mechanisms that satisfy this criterion include:
support for password entry by password managers to reduce memory need, and
copy and paste to reduce the cognitive burden of re-typing.

Note 3 (Added) (for non-web software)
Any passwords used to unlock underlying [platform software](https://w3c.github.io/wcag2ict/#platform-software) (running below the non-web software) are out of scope for this requirement since these are not under control of the non-web software’s author.

Note 4 (Added) (for non-web software)
There are cases where non-web software has an authentication process and no alternative or assistance mechanism is feasible, for example when entering a password when starting, powering on / turning on an ICT (device or otherwise). In such situations, it may not be possible for the non-web software to satisfy this success criterion.

Note 5 (Added) (for non-web software)
See also the [Comments on Closed Functionality](https://w3c.github.io/wcag2ict/#comments-on-closed-functionality).

**Comment by Lboniello** [comment]
Do we need all 5 notes or just note 3, 4, 5? @pday1

**Comment by pday1** [comment]
> Do we need all 5 notes or just note 3, 4, 5? [@pday1](https://github.com/pday1)

Hi @Lboniello I would think notes 2-5.

**Comment by pday1** [comment]
@Lboniello, having looked at this again, I think you are proposing adding notes 2-5 from 3.3.8 into the "applying" section of 3.3.9, which I agree with. 

However, you haven't proposed any new content for SC problematic for closed. I wonder if it would be appropriate to use the same text as for 3.3.8. 

This would result in the following proposed text. 

Applying SC 3.3.9 Accessible Authentication (Enhanced) to non-web documents and non-web software

> This applies directly as written, and as described in Intent from Understanding Success Criterion 3.3.9.
> 
> Note 2
> Examples of mechanisms that satisfy this criterion include:
> support for password entry by password managers to reduce memory need, and
> copy and paste to reduce the cognitive burden of re-typing.
> 
> Note 3 (Added) (for non-web software)
> Any passwords used to unlock underlying platform software (running below the non-web software) are out of scope for this requirement since these are not under control of the non-web software’s author.
> 
> Note 4 (Added) (for non-web software)
> There are cases where non-web software has an authentication process and no alternative or assistance mechanism is feasible, for example when entering a password when starting, powering on / turning on an ICT (device or otherwise). In such situations, it may not be possible for the non-web software to satisfy this success criterion.
> 
> NOTE (FOR NON-WEB SOFTWARE)
> See also the Comments on Closed Functionality.

Suggested content for SC in Success Criteria Problematic for Closed Functionality: 

> 3.3.9 Accessible Authentication (Enhanced) — There are situations where meeting this success criterion is problematic for ICT with closed functionality:
> 
> - Systems that are designed for shared use (such as in a public library) or have closed functionality might block mechanisms typically used to assist the user, such as copying authentication information from a password manager. Instead, an alternative authentication method might be needed, such as an identity card scanner.
> - Where standards for banking or security have authentication requirements that are regulated or strictly enforced, those requirements may be judged to take legal precedence over  Success Criterion 3.3.9 Accessible Authentication (Enhanced).

**Comment by Lboniello** [comment]
NEW CONTENT (will add to pull request)

Applying SC 3.3.9 Accessible Authentication (Enhanced) to non-web documents and non-web software

NEW CONTENT (will add to pull request)

 This applies directly as written, and as described in Intent from Understanding Success Criterion 3.3.9.

 <div class="note wcag2ict">Note 2
Examples of mechanisms that satisfy this criterion include: support for password entry by password managers to reduce memory need, and copy and paste to reduce the cognitive burden of re-typing.</div>

<div class="note wcag2ict">Note 3 (Added) (for non-web software) 
Any passwords used to unlock underlying platform software (running below the non-web software) are out of scope for this requirement since these are not under control of the non-web software’s author.</div>

<div class="note wcag2ict">Note 4 (Added) (for non-web software)
There are cases where non-web software has an authentication process and no alternative or assistance mechanism is feasible, for example when entering a password when starting, powering on / turning on an ICT (device or otherwise). In such situations, it may not be possible for the non-web software to satisfy this success criterion.</div>

<div class="note wcag2ict">NOTE (FOR NON-WEB SOFTWARE)
See also the Comments on Closed Functionality.</div>

Suggested content for SC in Success Criteria Problematic for Closed Functionality:

3.3.9 Accessible Authentication (Enhanced) — There are situations where meeting this success criterion is problematic for ICT with closed functionality: 
 * Systems that are designed for shared use (such as in a public library) or have closed functionality might block mechanisms typically used to assist the user, such as copying authentication information from a password manager. Instead, an alternative authentication method might be needed, such as an identity card scanner.
 * Where standards for banking or security have authentication requirements that are regulated or strictly enforced, those requirements may be judged to take legal precedence over Success Criterion 3.3.8 Accessible Authentication (Minimum).

**Comment by Lboniello** [comment]
Applying SC 3.3.9 Accessible Authentication (Enhanced) to non-web documents and non-web software
This applies directly as written, and as described in Intent from Understanding Success Criterion 3.3.9.

Note
Examples of mechanisms that satisfy this criterion include:
 support for password entry by password managers to reduce memory need, and copy and paste to reduce the cognitive burden of re-typing.

Note (Added) (for non-web software)
 Any passwords used to unlock underlying platform software (running below the non-web software) are out of scope for this requirement since these are not under control of the non-web software’s author.

Note (Added) (for non-web software)
There are cases where non-web software has an authentication process and no alternative or assistance mechanism is feasible, for example when entering a password when starting, powering on / turning on an ICT (device or otherwise). In such situations, it may not be possible for the non-web software to satisfy this success criterion.

 NOTE (FOR NON-WEB SOFTWARE)
 See also the Comments on Closed Functionality.


@pday1 is this ready to be committed?

**Comment by pday1** [comment]
Now in PR https://github.com/w3c/wcag2ict/pull/923
